kmip/pie/api.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class KmipClientAPI:
+    """
+    A simplified KMIP client interface for conducting KMIP operations.
+
+    The KmipClientAPI provides a simple external interface for various KMIP
+    operations and composes the bulk of the PyKMIP Pie API.
+    """
+    @abc.abstractmethod
+    def create(self, algorithm, length, *args, **kwargs):
+        """
+        Create a symmetric key on a KMIP appliance.
+
+        Args:
+            algorithm (CryptographicAlgorithm): An enumeration defining the
+                algorithm to use to generate the symmetric key.
+            length (int): The length in bits for the symmetric key.
+            args (various): Additional positional arguments for creating keys.
+            kwargs (various): Additional keyword arguments for creating keys.
+        """
+        pass
+
+    @abc.abstractmethod
+    def create_key_pair(self, algorithm, length, *args, **kwargs):
+        """
+        Create an asymmetric key pair on a KMIP appliance.
+
+        Args:
+            algorithm (CryptographicAlgorithm): An enumeration defining the
+                algorithm to use to generate the key pair.
+            length (int): The length in bits for the key pair.
+            args (various): Additional positional arguments for creating key
+                pairs.
+            kwargs (various): Additional keyword arguments for creating key
+                pairs.
+        """
+        pass
+
+    @abc.abstractmethod
+    def register(self, secret, *args, **kwargs):
```

Comment:
*panics* Umm, not `key` since `register` can be used for other objects.....I think I still prefer `secret`? But a `template` can be registered too and that's definitely not a `secret`. Ok, I'm fine with `managed_object`. Changing now...